feat(docs): add doc for public CSS references (#791)
Vonage/vivid-3: docs/design/design-tokens.md

![alias-tokens-explainer](https://user-images.githubusercontent.com/106529909/185866411-98c26728-8bec-4836-a440-a76469edd25d.png)

## Using design tokens

> Aside from the [color-related tokens](/designs/color-palette), you may use these ones to tweak your application appearance.

Reviewer: ```suggestion
Aside from the [color-related tokens](/designs/color-palette), you may use the following to customize your application appearance.
```

### Typography
#### Font families

Use `--vvd-font-family-upright` to specify a base proportional font.

Use `--vvd-font-family-monospace` to specify a base monospaced font.

> Unless explicitely stated otherwise, Vonage products should use the brand font `Spezia`. This is explained [in this repo](https://github.com/Vonage/spezia-webfont-kit).

Reviewer: and spezia mono?

- Type: `string`
- Default upright: `'Montserrat'`

> - Default monospace: `'Roboto Mono'`

Reviewer: should align with conventional docs where each members gets its own `type` and `default`

```html preview
<style>
  :root {
    --vvd-font-family-upright: Georgia;
    --vvd-font-family-monospace: Courier;
  }
</style>
<vwc-header>
  This header is using Georgia
</vwc-header>
<pre>/*
 * This block of code uses Courier
 */
</pre>
<vwc-button appearance="filled" label="I use Georgia too"></vwc-button>
```

#### Font size

> Use `--vvd-size-font-scale-base` to define the base font size for the widgets.

Reviewer: there's no guideline explicitly defines tweaking it in widgets. there's no actual knowledge or constraints on where this should be applied

> When the `.vvd-root` class is set on the root element (i.e., `<html>`), the default font size is reset to the user-agent's one (generally `16px`)

Reviewer: that's not necessarily true. depends if application consumes the css core. actually I can't seem to understand whether this paragraph is correct

and `--vvd-size-font-scale-base` set to `1rem`.

- Type: [`<length>`](https://developer.mozilla.org/en-US/docs/Web/CSS/length)
- Default: `undefined` or `'1rem'` (see above)

```html preview
<style>
  :root {
    --vvd-size-font-scale-base: 1.3rem;
  }
</style>
<vwc-header>
  This header's title is 30% larger than its parent
</vwc-header>
<pre>/*
 * This block of code too
 */
</pre>
<vwc-button appearance="filled" label="And the button label"></vwc-button>
```

#### Density

> Use `--vvd-size-density` to modify the spacing and padding of widgets. Allowed values are `-1` (condensed), `0` (default) and `1` (expanded).

Reviewer: ```suggestion
Use `--vvd-size-density` to modify the spacing and padding of widgets. Allowed values are `-1` (condensed), `0` (default), `1` and `2` (expanded).
```

> - Type: `-1` | `0` | `1`

Reviewer: ```suggestion
- Type: `-1` | `0` | `2`
```

![alias-tokens-explainer](https://user-images.githubusercontent.com/106529909/185866411-98c26728-8bec-4836-a440-a76469edd25d.png)

## Using design tokens

Aside from the [color-related tokens](/designs/color-palette), you may use these ones to tweak your application appearance.

### Typography
#### Font families

Use `--vvd-font-family-upright` to specify a base proportional font.

Use `--vvd-font-family-monospace` to specify a base monospaced font.

```html preview
<style>
  :root {
    --vvd-font-family-upright: Georgia;

>     --vvd-font-family-monospace: Courier;

Reviewer: it's a pickle. we have no component to expose this change. so this could be confusing for readers to understand why it's present here

Author: for monospace?

  }
</style>
<vwc-header>
  This header is using Georgia
</vwc-header>
<pre>/*
 * This block of code uses Courier
 */
</pre>
<vwc-button appearance="filled" label="I use Georgia too"></vwc-button>
```

#### Font size

Use `--vvd-size-font-scale-base` to define the base font size for the widgets. The default value in Vivid is `16px`.

> 

Reviewer: the reason this token is exposed is to enable authors to set typography to lean on relative units. maybe this is a right section to explain how that helps users expecting font sizing to follow their browser preference

Author: ok. the use I had in mind was for apps that provide controls to enlarge/shrink their text (without the use having to zoom).
if the main rationale is to say "as a user you can set a relative size" why don't we do it as a default?

Reviewer: our root's base font size is 16px, body increments by 1 in the typography scale (major second). the variable should be applied on the root

Reviewer: I'm not sure the issue with zoom-in on 14px in controls should be addressed by that variable. u could set the base to a value larger than 16 and result in larger than 14 in controls.
BUT, there should probably be an alternate typography tokens file dedicated for mobile

Reviewer: it's now practiced in #836

```html preview
<style>
  :root {

>     --vvd-size-font-scale-base: 1.3rem;

Reviewer: `1.3rem` from what? to set this to be deterministic let's add `font-size: 1rem` to `:root` as well. it'll default to the UA 16px or to user's browser preference in settings
<img width="723" alt="image" src="https://user-images.githubusercontent.com/15140652/200139471-4ffe959c-15a0-44e0-bbe6-91f08a5cf078.png">

Author: typography in CSS gives me headaches :)
it's 1.3rem from the default 16px mentioned above.

Reviewer: see #836

  }
</style>
<vwc-header>
  This header's title is 30% larger
</vwc-header>
<pre>/*
 * This block of code too
 */
</pre>
<vwc-button appearance="filled" label="And the button label"></vwc-button>
```

#### Density

Use `--vvd-size-density` to modify the spacing and padding of widgets. Allowed values are `-1` (condensed), `0` (default) and `1` (expanded).

```html preview
<style>
  :root {
    --vvd-size-density: -1;
  }
</style>
<vwc-header>
  This header is condensed
</vwc-header>
<pre>/*
 * This block of code is not affected
 */
</pre>
<vwc-button appearance="filled" label="But the button is"></vwc-button>
```

#### Styles preset

The following presets can be used to style the text in your application:

> - --vvd-typography-headline

Reviewer: not?
```suggestion
`--vvd-typography-headline`
```

Author: indeed!

- --vvd-typography-subtitle
- --vvd-typography-heading-1
- --vvd-typography-heading-2
- --vvd-typography-heading-3
- --vvd-typography-heading-4

> - --vvd-typography-base-condensed
> - --vvd-typography-base-condensed-bold
> - --vvd-typography-base-condensed-code
> - --vvd-typography-base
> - --vvd-typography-base-bold
> - --vvd-typography-base-code
> - --vvd-typography-base-extended
> - --vvd-typography-base-extended-bold
> - --vvd-typography-base-extended-code

Reviewer: sorted base to be first and dropping other code sizes. should consider removing their support so let's not expose for the time being
```suggestion
- --vvd-typography-base
- --vvd-typography-base-bold
- --vvd-typography-base-code
- --vvd-typography-base-condensed
- --vvd-typography-base-condensed-bold
- --vvd-typography-base-extended
- --vvd-typography-base-extended-bold
```

Author: yeah I changed the order because it felt more pleasing visually, but I'll remove that part anyway.

```html preview
<div style="font: var(--vvd-typography-headline)">headline</div>
<div style="font: var(--vvd-typography-subtitle)">subtitle</div>
<div style="font: var(--vvd-typography-heading-1)">heading-1</div>
<div style="font: var(--vvd-typography-heading-2)">heading-2</div>
<div style="font: var(--vvd-typography-heading-3)">heading-3</div>
<div style="font: var(--vvd-typography-heading-4)">heading-4</div>

> <div style="font: var(--vvd-typography-base-condensed)">base-condensed</div>
> <div style="font: var(--vvd-typography-base-condensed-bold)">base-condensed-bold</div>
> <div style="font: var(--vvd-typography-base-condensed-code)">base-condensed-code</div>
> <div style="font: var(--vvd-typography-base)">base</div>
> <div style="font: var(--vvd-typography-base-bold)">base-bold</div>
> <div style="font: var(--vvd-typography-base-code)">base-code</div>
> <div style="font: var(--vvd-typography-base-extended)">base-extended</div>
> <div style="font: var(--vvd-typography-base-extended-bold)">base-extended-bold</div>
> <div style="font: var(--vvd-typography-base-extended-code)">base-extended-code</div>

Reviewer: sorted base to be first and dropping other code sizes. should consider removing their support so let's not expose for the time being
```suggestion
<div style="font: var(--vvd-typography-base)">base</div>
<div style="font: var(--vvd-typography-base-bold)">base-bold</div>
<div style="font: var(--vvd-typography-base-code)">base-code</div>
<div style="font: var(--vvd-typography-base-condensed)">base-condensed</div>
<div style="font: var(--vvd-typography-base-condensed-bold)">base-condensed-bold</div>
<div style="font: var(--vvd-typography-base-extended)">base-extended</div>
<div style="font: var(--vvd-typography-base-extended-bold)">base-extended-bold</div>
```
